### PR TITLE
switch imx6ull to u-boot imx-2022.04

### PR DESCRIPTION
**Pull request by MrCry0**
Known issues:
- enabling PSCI breaks booting kernel. (temporarily disabled by not applying an overlay generated by op-tee).
- login service fails to start on a serial console.

### CODE REVIEW

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx6ullevk/lmp.cfg [hunk 1]
```diff
@@ -51,7 +51,7 @@ CONFIG_OPTEE=y
 CONFIG_OF_LIBFDT_OVERLAY=y
 CONFIG_OPTEE_TZDRAM_SIZE=0x02000000
 CONFIG_OPTEE_TZDRAM_BASE=0x9E000000
-CONFIG_OPTEE_SKIP_LOWLEVEL_INIT=y
+CONFIG_SKIP_LOWLEVEL_INIT=y
```
Comment (angolini) [inline meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx6ullevk/lmp.cfg:57]: just asking... do we need the same for meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx6ulevk/lmp.cfg?

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx6ullevk/lmp.cfg [hunk 2]
```diff
@@ -10,16 +10,16 @@ CONFIG_SPL_LOAD_FIT=y
 CONFIG_SPL_RSA=y
 CONFIG_SPL_LIBCOMMON_SUPPORT=y
 CONFIG_SPL_LIBGENERIC_SUPPORT=y
-CONFIG_SPL_SHA256_SUPPORT=y
-CONFIG_SPL_CRYPTO_SUPPORT=y
-CONFIG_SPL_HASH_SUPPORT=y
+CONFIG_SPL_SHA256=y
```
Comment (angolini) [inline meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx6ullevk/lmp.cfg:14]: Do we need the same for imx6ul mfgtools?

**Review by angolini — APPROVED** [review]

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx6ullevk/lmp.cfg (hunk 2 above)
Comment (MrCry0) [reply meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx6ullevk/lmp.cfg:14]: @angolini this set is for 6ull only. 6ul support is coming with another patchset.

File: meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx6ullevk/lmp.cfg (hunk 1 above)
Comment (MrCry0) [reply meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx6ullevk/lmp.cfg:57]: @angolini this set is for 6ull only. 6ul support is coming with another patchset.

Comment (angolini) [reply meta-lmp-bsp/recipes-bsp/u-boot/u-boot-fio-mfgtool/imx6ullevk/lmp.cfg:57]: ok

**Review by angolini — APPROVED** [review]

**Comment by quaresmajose** [comment]
@MrCry0 maybe is better to rebase the target main-nxp-bsp-update branch on top of main so we can drop some merged commits from this PR as in https://github.com/foundriesio/meta-lmp/compare/main...quaresmajose:meta-lmp:main-nxp-bsp-update?expand=1 (rebased on my repo)

**Comment by MrCry0** [comment]
> @MrCry0 maybe is better to rebase the target main-nxp-bsp-update branch on top of main so we can drop some merged commits from this PR as in https://github.com/foundriesio/meta-lmp/compare/main...quaresmajose:meta-lmp:main-nxp-bsp-update?expand=1 (rebased on my repo)

It is rebased. I just rewrote the fio's `main-nxp-bsp-update` so now this PR should look better :)

**Comment by MrCry0** [comment]
These commits are included into #862

**Pull request closed without merging**